Source/SpectrumAnalysis/KTSequentialTrackFinder.cc

```diff
@@ -329,225 +341,312 @@ namespace Katydid
-
-                         // if point matches this line: insert
-                         if (timeCondition and anyPointCondition)
-                         {
-                             KTDEBUG(stflog, "Matching conditions fullfilled");
-                             lineIt->AddPoint(*pointIt);
-                             (this->*fCalcSlope)(*lineIt);
-                             match = true;
-                             break;
-                         }
-                         // if this line consists of only one point so far, try again with different radius
-                         else if (lineIt->GetNPoints() == 1 and timeCondition and secondPointCondition)
-                         {
-                             KTDEBUG(stflog, "Trying initial-frequency-acceptance "<<fInitialFrequencyAcceptance);
-                             lineIt->AddPoint(*pointIt);
-                             (this->*fCalcSlope)(*lineIt);
-                             match = true;
-                             break;
-                         }
-                         // if not try next line
-                         else
-                         {
-                             ++lineIt;
-                         }
-                     }
-                 }//end of while loop. all existing lines were compared
-
-                 // if point was not picked up
-                 if (match == false)
-                 {
-                     //KTDEBUG(stflog, "Starting new line");
-
-                     KTSequentialLineData newLine;
-                     newLine.SetSlope( fInitialSlope );
-                     newLine.SetAcquisitionID( acqID );
-                     newLine.SetComponent( component );
-                     newLine.AddPoint(*pointIt);
-                     (this->*fCalcSlope)(newLine);
-                     fActiveLines.push_back(newLine);
-                     match = true;
-                 }
-             }
-         }
-         return true;
-     }
+        uint64_t acqID = kdTreeData.GetAcquisitionID();
+
+        // We need to be able to detect when we've moved from slice to slice
+        // So we define a threshold for delta-t, because time values might not be the exact same due to floating-point uncertainty.
+        // We'll use the bin width, which is tiny relative to the slice size and stride.
+        // 3-times the bin width should be large enough compared to uncertainty on the slice time and smaller than any reasonable stride.
+        double deltaTThreshold = 3. * fTimeBinWidth;
+
+        for (unsigned iComponent = 0; iComponent < nComponents; ++iComponent)
+        {
+            unsigned nBins = fMaxBin - fMinBin + 1;
+
+            const KTKDTreeData::SetOfPoints& allPoints = kdTreeData.GetSetOfPoints(iComponent);
+
+            // this set will collect the discriminated points sorted by power
+            STFDiscriminatedPowerSortedPoints points;
```

Comment:
Is this all points from the file? How did you make sure that they only get sorted by power within the same slice?

Reply:
`allPoints` is all points from the file.  At any given time `points` only holds the points from a single slice.  This is guaranteed by comparing the current point's (`pointIt`) time and the last point's time, and seeing if it changes by 3 time bins or more.  If it does, that means we've moved onto the next time slice.  That check is made at line 386, and we process and clear everything in `points` in that block.  Only after that do we deal with `pointIt` and add its information to `points`.